### DISCUSSION TITLE
session设置为database后，sessions表的user_id为空

### DISCUSSION THREAD

**Discussion by szyhbcn** [jqhph/dcat-admin#1995] [Q&A]
session设置为database后，登陆时sessions表的user_id为空，其它字段均正常.
尝试将DatabaseSessionHandler.php第222行return $this->container->make(Guard::class)->id();改为return Admin::user()->id ?? null;后，登陆系统时，user_id表可见用户ID。
这是配置问题还是系统问题？

**Comment by yiming0** [comment]
是 `dcat-admin` 的问题：在 `Dcat\Admin\Http\Middleware\Authenticate` ([source](https://github.com/jqhph/dcat-admin/blob/f8ef27cc4d6a79dc346f89d0efb925d4e28ee763/src/Http/Middleware/Authenticate.php#L23)) 中间件没有设置 Laravel 的用户验证驱动为 `'admin'` ，导致后续 `\Illuminate\Auth\SessionGuard` 获取不到正确的用户 ID。

内部原因：如你所说，当 `DatabaseSessionHandler` 尝试获取登录用户 ID 时  ([source](https://github.com/laravel/framework/blob/4e8e644f58cdacd9b330f4bc4fe39b4f58c97270/src/Illuminate/Session/DatabaseSessionHandler.php#LL221C59-L221C59))，其中 `$this->container->make(Guard::class)` Laravel 容器返回的 `\Illuminate\Auth\SessionGuard` 实例的属性 `name` 为默认值 `'web'` 而不是 `'admin'` ，导致 `user_id` 结果为 `null`。

**Reply by yiming0** [reply]
已提交修复 #2032
